### PR TITLE
Use a more general heuristic to identify message senders

### LINKED ISSUE #2: Bubbles don't appear on multiplayer servers
Hi, I'm playing on my friends server but the chat bubbles don't appear when a custom chat plugin is used:

When my friend removed his chat plugin the chat bubbles worked:

I'm also having this problem on a lot of multiplayer servers that are using custom chat plugins, it causes the talkbubbles to not work. Is there a way to fix this?

### LINKED ISSUE #7: enhanced player detection
The mod [Chat Heads](https://www.curseforge.com/minecraft/mc-mods/chat-heads) has non-UUID based "heuristics" to detect player heads which works flawlessly even on servers with custom chat plugins. Implementing this could solve #2 .
image
Chat Heads is also [open source](https://github.com/dzwdz/chat_heads) though [licensed under a GPL-like](https://github.com/dzwdz/chat_heads/blob/fabric-1.16.x/LICENSE) so I'm not sure you can directly copy the code but you can figure out exactly how this detection works to mimic it

### LINKED ISSUE #11: Not showing up the text bubble on the multiplayer
### Minecraft Version

1.19.x

### Mod Loader Version

1.19.4

### Mod Version

Talkbubbles-1.0.6+1.19.4

### Describe the Issue

When the join the server is not showing any textbubble on me and my friend but on the singleplayer is work fine 

### latest.log

[15:11:27] [main/INFO]: Loading Minecraft 1.19.4 with Fabric Loader 0.14.19
[15:11:27] [main/INFO]: Loading 58 mods:
	- cloth-config 10.0.96
	   \-- cloth-basic-math 0.6.1
	- fabric-api 0.81.1+1.19.4
	   |-- fabric-api-base 0.4.26+1e9487d2f4
	   |-- fabric-api-lookup-api-v1 1.6.27+504944c8f4
	   |-- fabric-biome-api-v1 13.0.8+348a9c64f4
	   |-- fabric-block-api-v1 1.0.7+e022e5d1f4
	   |-- fabric-blockrenderlayer-v1 1.1.36+c2e6f674f4
	   |-- fabric-client-tags-api-v1 1.0.17+504944c8f4
	   |-- fabric-command-api-v1 1.2.29+f71b366ff4
	   |-- fabric-command-api-v2 2.2.8+504944c8f4
	   |-- fabric-commands-v0 0.2.46+df3654b3f4
	   |-- fabric-containers-v0 0.1.56+df3654b3f4
	   |-- fabric-content-registries-v0 3.5.11+ae0966baf4
	   |-- fabric-convention-tags-v1 1.5.1+fe8721bef4
	   |-- fabric-crash-report-info-v1 0.2.16+aeb40ebef4
	   |-- fabric-data-generation-api-v1 11.4.2+504944c8f4
	   |-- fabric-dimensions-v1 2.1.47+7f87f8faf4
	   |-- fabric-entity-events-v1 1.5.15+504944c8f4
	   |-- fabric-events-interaction-v0 0.5.1+76ba65ebf4
	   |-- fabric-events-lifecycle-v0 0.2.54+df3654b3f4
	   |-- fabric-game-rule-api-v1 1.0.34+a1ccd7bff4
	   |-- fabric-item-api-v1 2.1.19+504944c8f4
	   |-- fabric-item-group-api-v1 3.0.7+043f9acff4
	   |-- fabric-key-binding-api-v1 1.0.34+504944c8f4
	   |-- fabric-keybindings-v0 0.2.32+df3654b3f4
	   |-- fabric-lifecycle-events-v1 2.2.17+1e9487d2f4
	   |-- fabric-loot-api-v2 1.1.29+75e98211f4
	   |-- fabric-loot-tables-v1 1.1.33+9e7660c6f4
	   |-- fabric-message-api-v1 5.1.3+504944c8f4
	   |-- fabric-mining-level-api-v1 2.1.41+49abcf7ef4
	   |-- fabric-models-v0 0.3.32+504944c8f4
	   |-- fabric-networking-api-v1 1.3.3+504944c8f4
	   |-- fabric-networking-v0 0.3.43+df3654b3f4
	   |-- fabric-object-builder-api-v1 7.0.5+504944c8f4
	   |-- fabric-particles-v1 1.0.25+f1e4495bf4
	   |-- fabric-recipe-api-v1 1.0.10+a1ccd7bff4
	   |-- fabric-registry-sync-v0 2.2.2+504944c8f4
	   |-- fabric-renderer-api-v1 2.2.7+ebc93ff3f4
	   |-- fabric-renderer-indigo 1.2.0+ebc93ff3f4
	   |-- fabric-renderer-registries-v1 3.2.40+df3654b3f4
	   |-- fabric-rendering-data-attachment-v1 0.3.30+afca2f3ef4
	   |-- fabric-rendering-fluids-v1 3.0.23+504944c8f4
	   |-- fabric-rendering-v0 1.1.43+df3654b3f4
	   |-- fabric-rendering-v1 2.1.3+504944c8f4
	   |-- fabric-resource-conditions-api-v1 2.3.2+e6c7d4eef4
	   |-- fabric-resource-loader-v0 0.11.4+5ade3c38f4
	   |-- fabric-screen-api-v1 1.0.47+3bd4ab0ff4
	   |-- fabric-screen-handler-api-v1 1.3.22+504944c8f4
	   |-- fabric-sound-api-v1 1.0.10+504944c8f4
	   |-- fabric-transfer-api-v1 3.2.0+80d07a0af4
	   \-- fabric-transitive-access-wideners-v1 3.1.1+b4a333d6f4
	- fabricloader 0.14.19
	- java 17
	- minecraft 1.19.4
	- modmenu 6.1.0-rc.2
	- talkbubbles 1.0.6
	- tlskincape 1.30
[15:11:27] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/C:/Users/TJ/AppData/Roaming/.minecraft/libraries/net/fabricmc/sponge-mixin/0.12.4+mixin.0.8.5/sponge-mixin-0.12.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[15:11:27] [main/INFO]: Compatibility level set to JAVA_16
[15:11:27] [main/INFO]: Compatibility level set to JAVA_17
[15:11:32] [Datafixer Bootstrap/INFO]: 180 Datafixer optimizations took 108 milliseconds
[15:11:34] [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[15:11:35] [Render thread/ERROR]: Failed to verify authentication
com.mojang.authlib.exceptions.InvalidCredentialsException: Status: 401
	at com.mojang.authlib.exceptions.MinecraftClientHttpException.toAuthenticationException(MinecraftClientHttpException.java:56) ~[authlib-3.18.38.1.jar:?]
	at com.mojang.authlib.yggdrasil.YggdrasilUserApiService.fetchProperties(YggdrasilUserApiService.java:156) ~[authlib-3.18.38.1.jar:?]
	at com.mojang.authlib.yggdrasil.YggdrasilUserApiService.<init>(YggdrasilUserApiService.java:55) ~[authlib-3.18.38.1.jar:?]
	at com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService.createUserApiService(YggdrasilAuthenticationService.java:161) ~[authlib-3.18.38.1.jar:?]
	at net.minecraft.class_310.method_31382(class_310.java:732) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.<init>(class_310.java:440) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:198) ~[Neat.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) ~[fabric-loader-0.14.19.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.19.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.19.jar:?]
Caused by: com.mojang.authlib.exceptions.MinecraftClientHttpException: Status: 401
	at com.mojang.authlib.minecraft.client.MinecraftClient.readInputStream(MinecraftClient.java:85) ~[authlib-3.18.38.1.jar:?]
	at com.mojang.authlib.minecraft.client.MinecraftClient.get(MinecraftClient.java:48) ~[authlib-3.18.38.1.jar:?]
	at com.mojang.authlib.yggdrasil.YggdrasilUserApiService.fetchProperties(YggdrasilUserApiService.java:129) ~[authlib-3.18.38.1.jar:?]
	... 8 more
[15:11:35] [Render thread/INFO]: Setting user: BlackFlame
[15:11:35] [Render thread/INFO]: [Indigo] Registering Indigo renderer!
[15:11:35] [Worker-Main-1/INFO]: Checking mod updates...
[15:11:35] [Render thread/INFO]: Backend library: LWJGL version 3.3.1 SNAPSHOT
[15:11:35] [Worker-Main-1/INFO]: Update available for 'modmenu@6.1.0-rc.2', (-> 6.2.2)
[15:11:36] [Render thread/INFO]: Reloading ResourceManager: vanilla, fabric (cloth-config, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-blockrenderlayer-v1, fabric-client-tags-api-v1, fabric-command-api-v1, fabric-command-api-v2, fabric-commands-v0, fabric-containers-v0, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-crash-report-info-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-events-lifecycle-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-keybindings-v0, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-tables-v1, fabric-message-api-v1, fabric-mining-level-api-v1, fabric-models-v0, fabric-networking-api-v1, fabric-networking-v0, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-renderer-registries-v1, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-rendering-v0, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-sound-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, modmenu, talkbubbles)
[15:11:37] [Realms Notification Availability checker #1/INFO]: Could not authorize you against Realms server: Invalid session id
[15:11:38] [Render thread/WARN]: Missing sound for event: minecraft:item.brush.brushing
[15:11:38] [Render thread/WARN]: Missing sound for event: minecraft:item.brush.brush_sand_completed
[15:11:38] [Render thread/WARN]: Missing sound for event: minecraft:item.goat_horn.play
[15:11:38] [Render thread/WARN]: Missing sound for event: minecraft:entity.goat.screaming.horn_break
[15:11:38] [Render thread/WARN]: Missing sound for event: minecraft:music.overworld.cherry_grove
[15:11:38] [Render thread/INFO]: OpenAL initialized on device OpenAL Soft on Speakers (BL100  PRO)
[15:11:38] [Render thread/INFO]: Sound engine started
[15:11:38] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/blocks.png-atlas
[15:11:38] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/signs.png-atlas
[15:11:38] [Render thread/INFO]: Created: 512x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
[15:11:38] [Render thread/INFO]: Created: 512x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
[15:11:38] [Render thread/INFO]: Created: 1024x1024x4 minecraft:textures/atlas/armor_trims.png-atlas
[15:11:38] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
[15:11:38] [Render thread/INFO]: Created: 64x64x4 minecraft:textures/atlas/decorated_pot.png-atlas
[15:11:38] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
[15:11:38] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
[15:11:38] [Render thread/WARN]: Shader rendertype_entity_translucent_emissive could not find sampler named Sampler2 in the specified shader program.
[15:11:38] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/particles.png-atlas
[15:11:38] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/paintings.png-atlas
[15:11:38] [Render thread/INFO]: Created: 128x128x0 minecraft:textures/atlas/mob_effects.png-atlas
[15:11:40] [Render thread/INFO]: [STDOUT]: [TLSkinCape] Current TLSkinCape version is 1.30
[15:11:46] [Render thread/INFO]: Connecting to bayanihan.minecraft.best, 25565
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 9ae95ae0-9bf8-3f1a-a050-81874a68d352
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 00000000-0000-0000-0009-01fdd6f25594
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player a845308a-2521-35e0-8dfe-1e0a2c0b2b45
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 00000000-0000-0000-0009-01f12a546196
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 9c924040-129a-3df4-9ceb-4435ab5f6975
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 009b9993-f3ad-4f4f-bac4-5fadd88eb75d
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 884d13a9-c98e-3c47-8e10-6c4135c935e7
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 2f2fbc49-39c6-32f0-bcf5-da4cdfe1b356
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 00000000-0000-0000-0009-01f88d82ff90
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 00000000-0000-0000-0009-01f90a5861d3
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 6ad3c00b-a377-3ba5-b7c6-b4abf7870b9c
[15:11:48] [Render thread/WARN]: Ignoring player info update for unknown player 9ae95ae0-9bf8-3f1a-a050-81874a68d352
[15:11:48] [Render thread/INFO]: [System] [CHAT] Welcome to the server.
[15:11:48] [Render thread/WARN]: Received packet for unknown team EBlackFlame: team action: REMOVE, player action: null
[15:11:48] [Render thread/INFO]: [System] [CHAT] [+] BlackFlame
[15:11:48] [Render thread/INFO]: [System] [CHAT] Your login session has been continued.
[15:11:48] [Render thread/INFO]: Loaded 511 advancements
[15:11:48] [Render thread/WARN]: Unknown custom packed identifier: labymod3:main
[15:11:48] [Render thread/INFO]: [System] [CHAT] 
[15:11:48] [Render thread/INFO]: [System] [CHAT]  SECURITY QUESTIONS
[15:11:48] [Render thread/INFO]: [System] [CHAT]  Perform the command /2fa squestion add to register your
[15:11:48] [Render thread/INFO]: [System] [CHAT]  security questions. By doing so, you
[15:11:48] [Render thread/INFO]: [System] [CHAT]  decrease the chance of being hacked.
[15:11:48] [Render thread/INFO]: [System] [CHAT] 
[15:11:48] [Render thread/INFO]: [System] [CHAT]   Confirm read      Remind me later  
[15:11:48] [Render thread/INFO]: [System] [CHAT] 
[15:11:49] [Render thread/INFO]: calculated cape performance:TEXTURE_272, middle time: 4.0
[15:11:49] [Render thread/WARN]: Received passengers for unknown entity
[15:11:49] [Render thread/INFO]: [System] [CHAT] [+] BlackFlame
[15:11:49] [Render thread/INFO]: [System] [CHAT] You have 8 sites to vote on still! do /vote to vote!
[15:11:49] [Render thread/INFO]: can use texture size:TEXTURE_272, middle time: 3.1
[15:11:50] [Render thread/INFO]: [System] [CHAT] [TimedRewards] You have unclaimed rewards! Do /rewards and claim them.
[15:11:50] [Render thread/INFO]: textures 'CREATIVE' was added, skin: true,cape: true, max waiting: 16 ,during : 327 
[15:11:52] [Render thread/INFO]: [System] [CHAT] [Guild MOTD] OMEN
[15:11:56] [Render thread/WARN]: Warning! Mod tlskincape has a broken icon, loading default icon
[15:12:01] [Render thread/INFO]: [System] [CHAT] [0MEN]? CAD ? a
[15:12:43] [Render thread/INFO]: [System] [CHAT] [0MEN]? CAD ? aaa
[15:12:56] [Render thread/WARN]: Ignoring player info update for unknown player fd8237c4-0b99-3934-aefd-30beca720c0b
[15:12:56] [Render thread/WARN]: Ignoring player info update for unknown player fd8237c4-0b99-3934-aefd-30beca720c0b
[15:12:56] [Render thread/WARN]: Received packet for unknown team EKayl: team action: REMOVE, player action: null
[15:12:56] [Render thread/INFO]: [System] [CHAT] [+] Kayl
[15:13:23] [Render thread/INFO]: [System] [CHAT] [-] Kayl
[15:13:24] [Render thread/WARN]: Ignoring player info update for unknown player fd8237c4-0b99-3934-aefd-30beca720c0b
[15:14:09] [Render thread/INFO]: [System] [CHAT] You have reached the maximum number of uses (1).
[15:14:09] [Render thread/INFO]: [System] [CHAT] [+] BlackFlame
[15:14:39] [Render thread/INFO]: Stopping!

### Do you use any other mods except the required ones?

Modmenu

### LINKED ISSUE #12: Bubbles don't work with EssentialsXChat
### Minecraft Version

1.19.x

### Mod Loader Version

0.14.19

### Mod Version

1.0.6

### Describe the Issue

When chatting on my server with EssentialsXChat, the bubbles don't show up. Here's a screenshot:

I tried removing my nickname entirely, and it still didn't work:

I should note that they work fine in singleplayer.

### latest.log

N/A, no crash issues

### Do you use any other mods except the required ones?

Doesn't work with all other mods disabled either.

### LINKED ISSUE #15: Support for 1.20.1 is needed
### Minecraft Version

1.20.x

### Mod Loader Version

0.14.21

### Mod Version

1.0.6

### Describe the Issue

Only 1.20 is supported but not 1.20.1

### latest.log

-

### Do you use any other mods except the required ones?

_No response_
javaw_ScIOLoVYDc

### PR DESCRIPTION
**Pull request by belcar-s**
This would fix #2 and others 😊.

<details>
<summary>Others</summary>

Fixes #7
Fixes #11
Fixes #12
Fixes #15

</details>

Steve speaking on a vanilla server. A chat bubble appears above their head.
Alex speaking on Hypixel. A chat bubble appears above their head.

### CODE REVIEW

**Comment by belcar-s** [comment]
🎵 I get so lonely —can't let just anybody hold me! 🎶🎤

File: src/main/java/net/talkbubbles/mixin/ChatHudMixin.java [hunk 1]
```diff
@@ -40,15 +40,20 @@ public class ChatHudMixin {
...
+
             if (!TalkBubbles.CONFIG.showOwnBubble)
                 list.remove(client.player);
             for (int i = 0; i < list.size(); i++)
-                if (list.get(i).getUuid().equals(extractSender(message))) {
+                if (list.get(i).getUuid().equals(senderUUID)) {
                     String stringMessage = message.getString();
-                    stringMessage = stringMessage.replace("<" + StringUtils.substringBetween(stringMessage, "<", ">") + "> ", "");
+                    stringMessage = stringMessage.replaceFirst("[\\s\\S]*" + detectedSenderName + "([^\\w§]|(§.)?)+\\s+", "");
+                    TalkBubbles.LOGGER.error(stringMessage);
```
Comment (Globox1997) [inline src/main/java/net/talkbubbles/mixin/ChatHudMixin.java]: You want to error log all messages?

Comment (belcar-s) [reply src/main/java/net/talkbubbles/mixin/ChatHudMixin.java]: Oops. Sorry. That was debugging.

**Comment by Globox1997** [comment]
Thanks for the great pr :) appreciate it!

**Pull request merged**